    public JDBCTransactionState( DataSource pool) throws IOException{
        try {
            connection = pool.getConnection();
            connection.setAutoCommit( false );                                    
        } catch (SQLException e) {
            connection = null;
            // This has not harmed the Transaction yet
            // so we don't need to force a rollback :-)
            throw new DataSourceException("Transaction could not acquire connection", e );
        }
    }

